### ISSUE TITLE
after moving the disk of a vm with vmotion, terraform cant find it anymore

### ISSUE THREAD

**Issue by mjrider** [terraform-providers/terraform-provider-vsphere#131]
### Terraform Version
Terraform v0.10.2
vsphere provider from git/master

### Affected Resource(s)
Please list the resources as a list, for example:
- vsphere_virtual_machine

### Terraform Configuration Files
```hcl
resource "vsphere_virtual_machine" "testvm4" {
	name          = "testvm4"
	vcpu          = 1
	memory        = 1024
	domain        = "example.org"
	datacenter    = "${var.vsphere_datacenter}"
	cluster       = "${var.vsphere_cluster}"
	resource_pool = "${var.vsphere_cluster}/${var.vsphere_resourcepool}"

	disk {
		datastore = "Shared Storages/DatastoreCluster01/pcc-004828"
		type = "thin"
		size = 2
                controller_type = "scsi-paravirtual"
		name = "disk1"
	}
}
```

### Debug Output
https://gist.github.com/mjrider/b114f376f1403eee350bfa2294a8cfa1

### Expected Behavior
What should have happened?
No changes in terraform plan, but with the current state of storage cluster support, a change that the datastore has changed

### Actual Behavior
the disk is not recognised:
**disk.#:                                 "0" => "1"**
```
  ~ vsphere_virtual_machine.testvm4
      disk.#:                                 "0" => "1"
      disk.1031547471.bootable:               "" => ""
      disk.1031547471.controller_type:        "" => "scsi-paravirtual"
      disk.1031547471.datastore:              "" => "Shared Storages/DatastoreCluster01/pcc-004828"
      disk.1031547471.iops:                   "" => ""
      disk.1031547471.keep_on_remove:         "" => ""
      disk.1031547471.key:                    "" => "<computed>"
      disk.1031547471.name:                   "" => "disk1"
      disk.1031547471.size:                   "" => "2"
      disk.1031547471.template:               "" => ""
      disk.1031547471.type:                   "" => "thin"
      disk.1031547471.uuid:                   "" => "<computed>"
      disk.1031547471.vmdk:                   "" => ""
```

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

### Important Factoids
the problem lies here:
https://github.com/terraform-providers/terraform-provider-vsphere/blob/master/vsphere/resource_vsphere_virtual_machine.go#L1041

**Comment by mjrider** [comment]
this might be a duplicate of https://github.com/terraform-providers/terraform-provider-vsphere/issues/49

**Comment by vancluever** [comment]
Thanks for the report @mjrider! Adding this to the TODO to look at getting fixed.

**Comment by vancluever** [comment]
Hey @mjrider - this should have been fixed with 1.0.0. If something like this comes up again, can you open a new issue?

Thanks!
